### ISSUE TITLE
DevicesTest tests fail when the hub contains devices with complex device twins

### ISSUE THREAD

**Issue by dluc** [Azure/iothub-manager-java#38]
# Type of issue
- [x] Bug
- [ ] New feature
- [ ] Enhancement

# Description
Some tests in `DevicesTest` fail when the IoT Hub used contains device twins containing more than 5 levels. The tests should not be affected by existing devices.

**Comment by zhang-hua** [comment]
This is caused by the limitation of current Java implementation of TwinParser which defines [MAX_MAP_LEVEL=5](https://github.com/Azure/azure-iot-sdk-java/blob/master/deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/TwinParser.java#L27) and throw IllegalArgumentException if level is greater than 5. 
I have created an issue [158](https://github.com/Azure/azure-iot-sdk-java/issues/158) for JAVA SDK.

Here is an example of failing twin reported property(4 levels based on document description):
```
"reported": { 
        "Protocol": "MQTT",
        "SupportedMethods": "Reboot,FirmwareUpdate,EmergencyValveRelease,IncreasePressure",
        "Telemetry": {                            -> Level 1
          "chiller-sensors;v1": {              -> Level 2
            "Interval": "00:00:10",
            "MessageTemplate": "{\"temperature\":${temperature},\"temperature_unit\":\"${temperature_unit}\",\"humidity\":${humidity},\"humidity_unit\":\"${humidity_unit}\",\"pressure\":${pressure},\"pressure_unit\":\"${pressure_unit}\"}",
            "MessageSchema": {             -> Level 3
              "Name": "chiller-sensors;v1",
              "Format": "JSON",
              "Fields": {                             -> Level 4
                "temperature": "Double", 
                "temperature_unit": "Text",
                "humidity": "Double",
                "humidity_unit": "Text",
                "pressure": "Double",
                "pressure_unit": "Text"
              }
            }
          }
        }
```

**Comment by dluc** [comment]
@zhang-hua can you make the test resilient to this issue? we're going to share resources between Java and .NET very often, and we cannot afford to have broken builds when there is no bug in the code

**Comment by zhang-hua** [comment]
Fixed in PR #64 and close it.
